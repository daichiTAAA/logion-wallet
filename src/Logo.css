.logo {
    height: 200px;
    background-color: black;
    text-align: center;
    padding-top: 10px;
    padding-bottom: 10px;
}

.logo > img {
    height: 180px;
}
