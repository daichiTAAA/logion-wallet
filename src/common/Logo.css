.Logo {
    padding: 40px;
    padding-right: 20px;
    height: 160px;
    z-index: 1;
}

.Logo .image-and-slogan {
    display: flex;
    height: 80px;
}

.Logo .image {
    height: 100%;
    margin-right: 20px;
}

.Logo .image img {
    height: 100%;
}

.Logo .slogan {
    display: flex;
    font-size: 14px;
    flex-direction: column;
    justify-content: flex-end;
}
