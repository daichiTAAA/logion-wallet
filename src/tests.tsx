import { ReactElement } from 'react';
import { createRenderer } from 'react-test-renderer/shallow';
import reactRenderer, { ReactTestRenderer, ReactTestRendererJSON, act as reactAct } from 'react-test-renderer';

export function render(element: ReactElement): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
    const component = reactRenderer.create(element);
    return component.toJSON();
}

export function renderer(element: ReactElement): ReactTestRenderer {
    return reactRenderer.create(element);
}

export function shallowRender<E extends ReactElement>(element: ReactElement): E {
    const renderer = createRenderer();
    renderer.render(element);
    return renderer.getRenderOutput();
}

export function mockAccount(address: string, name: string) {
    return {
        address,
        meta: {
            name
        }
    };
}

export { reactAct as act };
