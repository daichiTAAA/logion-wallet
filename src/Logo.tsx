import React from 'react';
import './Logo.css';

export default function Component() {
    return (
        <div className="logo">
            <img src={process.env.PUBLIC_URL + "/logo.png"} alt="logo" />
        </div>
    );
}
