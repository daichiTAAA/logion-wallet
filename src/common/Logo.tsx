import { useCommonContext } from './CommonContext';

import './Logo.css';

export interface Props {
    shadowColor: string,
}

export default function Logo(props: Props) {
    const { colorTheme } = useCommonContext();

    let fileName;
    if(colorTheme.type === 'dark') {
        fileName = 'logo.png';
    } else {
        fileName = 'logo_black.png';
    }

    return (
        <div
            className="Logo"
            style={{
                boxShadow: `-25px 5px 25px ${props.shadowColor}`,
            }}
        >
            <div className="image-and-slogan">
                <div className="image">
                    <img src={process.env.PUBLIC_URL + "/" + fileName} alt="logo" />
                </div>
                <div className="slogan">
                    <div className="text">
                        One blockchain<br/>
                        to trust them all
                    </div>
                </div>
            </div>
        </div>
    );
}
