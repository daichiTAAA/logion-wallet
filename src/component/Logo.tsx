import React from 'react';

import { BackgroundAndForegroundColors } from './ColorTheme';

import './Logo.css';

export interface Props extends BackgroundAndForegroundColors {
    logoShadow: string,
}

export default function Logo(props: Props) {

    return (
        <div
            className="Logo"
            style={{
                backgroundColor: props.background,
                color: props.foreground,
                boxShadow: `0 5px 25px ${props.logoShadow}`,
            }}
        >
            <div className="image-and-slogan">
                <div className="image">
                    <img src={process.env.PUBLIC_URL + "/logo.png"} alt="logo" />
                </div>
                <div className="slogan">
                    <div className="text">
                        One blockchain<br/>
                        to trust them all
                    </div>
                </div>
            </div>
        </div>
    );
}
