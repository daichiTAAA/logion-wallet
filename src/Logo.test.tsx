import React from 'react';
import Logo from './Logo';
import { render } from './tests';

test('renders', () => {
    const tree = render(<Logo />);
    expect(tree).toMatchSnapshot();
});
