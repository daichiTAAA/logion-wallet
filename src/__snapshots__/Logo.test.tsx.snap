// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="logo"
>
  <img
    alt="logo"
    height={100}
    src="/logo_black.png"
  />
</div>
`;
